src/subcommands/wallet/mod.rs

```diff
@@ -242,9 +243,9 @@ impl<'a> WalletSubCommand<'a> {
                 let payload = AddressPayload::from_pubkey_hash(hash160.clone());
                 lock_hashes.push(Script::from(&payload).calc_script_hash());
             }
-            last_change_address.payload().clone()
+            (last_change_address.payload().clone(), path_map.get(&change_last))
```

Comment:
It is an error if we don't find it. I think it is also always the last thing in the vector so we could just index that and do an assert.